gc/src/gc.rs

```diff
@@ -146,39 +146,39 @@ impl<
         // This relies on GcBox being #[repr(C)].
         let gcbox_layout = header_layout.extend(value_layout).unwrap().0.pad_to_align();
 
+        // Allocate the GcBox in a way that's compatible with Box,
+        // since the collector will deallocate it via
+        // Box::from_raw.
+        let gcbox_addr = unsafe { alloc(gcbox_layout) };
```

Comment:
issue: I'm not a fan of having each line be `unsafe` when the whole operation is a tied-together unsafe operation. Was this required by the edition change? I think this is more churn than I want in the unsafe code.

Reply:
Please do, thanks.